packages/arb-token-bridge-ui/src/token-bridge-sdk/OftV2TransferStarter.ts

```diff
@@ -158,14 +201,76 @@ export class OftV2TransferStarter extends BridgeTransferStarter {
     return contract.functions.approve(spender, constants.MaxUint256) // Eth USDT will need MAX approval since that cannot be changed afterwards
   }
 
-  // for OFT, we don't have functions for gas estimates, `sendQuote` method tells us the fees directly
-  public async transferEstimateGas() {
-    return undefined
+  public async transferEstimateGas({
+    amount,
+    signer,
+    destinationAddress
+  }: TransferEstimateGasProps) {
+    await this.validateOftTransfer()
+
+    const oftContract = this.getOftAdapterContract(signer)
+
+    const isDepositMode = isDepositModeUtil({
+      sourceChainId: await getChainIdFromProvider(this.sourceChainProvider),
+      destinationChainId: await getChainIdFromProvider(
+        this.destinationChainProvider
+      )
+    })
+
+    if (!this.sourceChainErc20Address) {
+      throw Error('OFT token address not found')
+    }
+
+    const allowance = await fetchErc20Allowance({
+      address: this.sourceChainErc20Address,
+      provider: this.sourceChainProvider,
+      owner: await signer.getAddress(),
+      spender: oftContract.address
+    })
+    if (allowance.lt(amount)) {
+      /**
+       * Default to hardcoded values based on sample of transactions:
+       *
+       * Arb1:
+       * https://arbiscan.io/tx/0xced6a1a14d42678f35c689756063021e27799e00a671522d09207d582184180c
+       * https://arbiscan.io/tx/0xb4bc8435131ffc489cd60060cd1eabdcb866cf38ffe9f1a09a818d515691d087
+       * https://arbiscan.io/tx/0x8ca579c757c36a66c5080524f215888b37e28d5ac00ed57ccd2248d6652eb72e
+       *
+       * Mainnet:
+       * https://etherscan.io/tx/0xbd2f476acb0d78817a8222da1fdcd3409aeb82b72b8e393780e1867c4bf5d010
+       * https://etherscan.io/tx/0xd8093e91850c50517b808510c30918cc79e32768c561d5f3dbfdb1398cd954ce
+       * https://etherscan.io/tx/0x3836f1f76333853e69ed975f412afdb7337cb3c4ced636bcdb8dfebc720b75a4
+       * https://etherscan.io/tx/0xe1d717d5063bf55742af9aef7a5600dcd8c0bd2553deca25e52e7bb2a48e65b6
+       */
+      const gasEstimate = isDepositMode
+        ? BigNumber.from(600_000)
```

Comment:
i know you have added a buffer but still let's bump these by 30% to be super safe.